lib/global_uid/base.rb

```diff
@@ -164,10 +174,21 @@ def self.get_connections(options = {})
       with_connections {}
     end
 
+    def self.validate_increment!(new_record_id, connection)
+      server = self.servers.find { |s| connection.current_database.include?(s[:name]) }
+      server[:allocated_ids] << new_record_id
+
+      valid_steps = server[:allocated_ids].each_cons(2).all? do |first_identifier, second_identifier|
+        (second_identifier - first_identifier) % self.global_uid_options[:increment_by] == 0
```

Comment:
It should never happen, but one thing this won't catch is if `first_identifier == second_identifier`

It might be faster to cache `self.global_uid_options[:increment_by]` in a local var

Reply:
>  but one thing this won't catch is if `first_identifier == second_identifier`

You're right. I was operating under the assumption that the database uniqueness constraint will be raised, I'll add a safeguard.